addon_service/management/commands/migrate_authorized_account.py

```diff
@@ -233,55 +233,59 @@ def migrate_for_user(
+            account = AuthorizedAccount(
+                display_name=service_name.capitalize(),
+                int_authorized_capabilities=(
+                    AddonCapabilities.UPDATE | AddonCapabilities.ACCESS
+                ).value,
+                account_owner=account_owner[0],
+                external_service=external_service,
+                credentials=credentials,
+                external_account_id=osf_account.provider_id,
             )
-            return
-        user_uri = f"{OSF_BASE}/{user.guid}"
-        account_owner = UserReference.objects.get_or_create(user_uri=user_uri)
-        account = AuthorizedAccount(
-            display_name=service_name.capitalize(),
-            int_authorized_capabilities=(
-                AddonCapabilities.UPDATE | AddonCapabilities.ACCESS
-            ).value,
-            account_owner=account_owner[0],
-            external_service=external_service,
-            credentials=credentials,
-            external_account_id=osf_account.provider_id,
-        )
 
-        if external_service.credentials_format == CredentialsFormats.OAUTH2:
-            token_metadata = OAuth2TokenMetadata(
-                refresh_token=osf_account.refresh_token,
-                access_token_expiration=osf_account.expires_at,
-                authorized_scopes=external_service.supported_scopes,
-            )
-            token_metadata.save(full_clean=False)
-            account.oauth2_token_metadata = token_metadata
+            if external_service.credentials_format == CredentialsFormats.OAUTH2:
+                token_metadata = OAuth2TokenMetadata(
+                    refresh_token=osf_account.refresh_token,
+                    access_token_expiration=osf_account.expires_at,
+                    authorized_scopes=external_service.supported_scopes,
+                )
+                token_metadata.save(full_clean=False)
+                account.oauth2_token_metadata = token_metadata
 
-        if api_url := self.get_api_base_url(external_service, osf_account):
-            try:
-                account.api_base_url = api_url
-            except ValidationError:
-                print(api_url)
+            if api_url := self.get_api_base_url(external_service, osf_account):
+                try:
+                    account.api_base_url = api_url
+                except ValidationError:
+                    print(api_url)
 
-        account.save(full_clean=False)
+            account.save(full_clean=False)
+            account_map[osf_account.id] = account
 
         for node_settings in node_settings_set:
+            node_guid = get_node_guid(node_settings.owner_id)
             resource_reference = ResourceReference.objects.get_or_create(
-                resource_uri=f"{OSF_BASE}/{get_node_guid(node_settings.owner_id)}"
+                resource_uri=f"{OSF_BASE}/{node_guid}"
             )[0]
+            base_account = account_map.get(node_settings.external_account.id)
```

Comment:
May need to `continue` here `if not base_account`, as a consequence of any `CredentialException` above